gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java

```diff
@@ -306,7 +306,10 @@ public B readTimeout(int readTimeout) {
   protected ServiceOptions(Class<? extends ServiceFactory<ServiceT, OptionsT>> serviceFactoryClass,
       Class<? extends ServiceRpcFactory<ServiceRpcT, OptionsT>> rpcFactoryClass,
       Builder<ServiceT, ServiceRpcT, OptionsT, ?> builder) {
-    projectId = checkNotNull(builder.projectId != null ? builder.projectId : defaultProject());
+    projectId = builder.projectId != null ? builder.projectId : defaultProject();
+    if (projectIdRequired()) {
+      checkNotNull(projectId);
```

Comment:
Sorry, should have mentioned that earlier.
Because we actually may accept null in some cases I think we should use checkArgument and use a nice message such as "This service requires a projectId"...